nixos/tests/budgie.nix

```diff
@@ -0,0 +1,51 @@
+import ./make-test-python.nix ({ pkgs, lib, ... }: {
+  name = "budgie";
+
+  meta = with lib; {
+    maintainers = [ maintainers.federicoschonborn ];
+  };
+
+  nodes.machine = { ... }: {
+    imports = [
+      ./common/user-account.nix
+    ];
+
+    services.xserver.enable = true;
+
+    services.xserver.displayManager = {
+      lightdm.enable = true;
+      autoLogin = {
+        enable = true;
+        user = "alice";
+      };
+    };
+
+    services.xserver.desktopManager.budgie.enable = true;
+  };
+
+  testScript = { nodes, ... }:
+    let
+      user = nodes.machine.users.users.alice;
+    in
+    ''
+      with subtest("Wait for login"):
+          machine.wait_for_x()
+          machine.wait_for_file("${user.home}/.Xauthority")
+          machine.succeed("xauth merge ${user.home}/.Xauthority")
+
+      with subtest("Check that logging in has given the user ownership of devices"):
+          machine.succeed("getfacl -p /dev/snd/timer | grep -q ${user.name}")
+
+      with subtest("Check if Budgie session components actually start"):
+          machine.wait_until_succeeds("pgrep budgie-wm")
+          machine.wait_for_window("budgie-wm")
```

Comment:
Hmmm, this step (actually grepping for `budgie-wm` in `xwininfo -root -tree`) fails when rebased to the GNOME 44 branch but the VM starts and seems to work. By using mutter 43 the test runs fine.

Is this expected? 

![](https://user-images.githubusercontent.com/20080233/229355645-57f5f710-a6b0-4f79-838a-35a66ddd3f55.png)

Reply:
(Commits used for testing: https://github.com/bobby285271/nixpkgs/commits/budgie-mutter-44, there is an unrelated fprintd test [failure](https://gitlab.freedesktop.org/libfprint/fprintd/-/issues/151) you want to ignore when testing).

Reply:
Okay I downloaded a Fedora 38 Beta image, the `xwininfo -root -tree` output looks fine with mutter 44-beta-3.fc38 (https://src.fedoraproject.org/rpms/mutter/commits/f38)

![](https://user-images.githubusercontent.com/20080233/229359917-747487d5-f435-433f-87a9-63b7c5373f42.png)

But after upgrading mutter to 44.0-1.fc38, budgie-wm also disappears from the output

![](https://user-images.githubusercontent.com/20080233/229360173-404e0842-541f-42ad-8a67-740122073d90.png)

So this is probably not NixOS-specific issue. If this is an expected behavior, we probably can just  remove this line (or maybe test something that can still pass with GNOME 44). If this is not expected behavior, not sure what to do then :upside_down_face:

Reply:
I switched from checking for `budgie-wm` to checking for `budgie-daemon`.